src/Repository/RunningMateRequestRepository.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\ApplicationRequest\RunningMateRequest;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\ORM\Query\Expr\Join;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+class RunningMateRequestRepository extends ServiceEntityRepository
+{
+    public function __construct(RegistryInterface $registry)
+    {
+        parent::__construct($registry, RunningMateRequest::class);
+    }
+
+    public function findForReferent(Adherent $referent): array
+    {
+        if (!$referent->isReferent()) {
+            return [];
+        }
+
+        return $this->createQueryBuilder('r')
+            ->select('r', 'CASE WHEN a.id IS NOT NULL THEN 1 ELSE 0 END AS isAdherent')
```

Comment:
`createQueryBuilder('r')` already adds `from()` and `select('r')`
here you could just call `addSelect('CASE ...')`